sound/soc/codecs/rt1308-sdw.c

```diff
@@ -571,6 +590,29 @@ static int rt1308_sdw_hw_params(struct snd_pcm_substream *substream,
 		return retval;
 	}
 
+	return retval;
+
+capture:
+	direction = SDW_DATA_DIR_TX;
+	num_channels = 2;
+
+	stream_config.frame_rate = params_rate(params);
+	stream_config.ch_count = num_channels;
+	stream_config.bps = snd_pcm_format_width(params_format(params));
+	stream_config.direction = direction;
+
+	port_config_feedback[0].ch_mask = BIT(0);
+	port_config_feedback[0].num = 2; /* DP2 */
+	port_config_feedback[1].ch_mask = BIT(1);
+	port_config_feedback[1].num = 4; /* DP4 */
```

Comment:
@plbossart Thanks for this commit to explain how to config the multi data ports with one DAI link.
I think something I don't explain clearly in #2513 
The feedback channels are two channels for each port.
Due to RT1308 is a stereo amplifier, it will protect the left and right speakers individually.
Left IV => using DP2 to feedback ch0 (I data) and ch1 (V data) for left speaker.
Right IV => using DP4 to feedback ch0 (I data) and ch1 (V data) for right speaker.
Therefore, the total channels are four channels like below.

```
	num_channels = 4;

	port_config_feedback[0].ch_mask = BIT(0) | BIT(1);
	port_config_feedback[0].num = 2; /* DP2 */
	port_config_feedback[1].ch_mask = BIT(2) | BIT(3);
	port_config_feedback[1].num = 4; /* DP4 */
```

Reply:
@shumingfan  waiting for your result. I also had such idea but I didn't get a answer from cadance's spec.

Reply:
@shumingfan I get the point for the case with a a single amp, we would get 4 channels. Your change makes sense. I have to see though how we would deal with 4 channels on the Intel side, I believe we would need 2 ports.

Also what happens in the case where there are two amplifiers in the system, and each of them controls a tweeter and woofer. How many feedback channels would we get from each amplifier?

Reply:
@plbossart It will have 4 feedback channels from each amplifier. 
Hence, there are eight feedback channels when using two amplifiers in the system.
The amp protection needs to protect each speaker whether it belongs to the tweeter or woofer.

Reply:
> @shumingfan waiting for your result. I also had such idea but I didn't get a answer from cadance's spec.

@RanderWang I tested the feedback function with the approach of this PR.
First, I tried to record the IV data from DP2 only. I can see the ch0 and ch1 have the captured data.
Then, I tried to record the IV data from DP2/4 both. There are captured data with ch0 and ch1 only. 
The ch2 and ch3 are zero.
I checked the MIPI-defined registers of DP2 and DP4 which had been configured while the test.
The DP4_ChannelEN doesn't enable. I think there is a problem there.

Reply:
@shumingfan if the DP4 registers were configured, but only the DP4_ChannelEN is not set, this points to a missing loop in the bank switch where we only deal with a single port?

Reply:
> @shumingfan if the DP4 registers were configured, but only the DP4_ChannelEN is not set, this points to a missing loop in the bank switch where we only deal with a single port?

@plbossart I found the root cause. The DP2/4_ChannelEn only supports channel1/2.
The driver needs to change the ch_mask to 0x3 for DP2/4 both.
Now, I could get the captured data of four channels.

```
num_channels = 4;

port_config_feedback[0].ch_mask = BIT(0) | BIT(1);
port_config_feedback[0].num = 2; /* DP2 */
port_config_feedback[1].ch_mask = BIT(0) | BIT(1);
port_config_feedback[1].num = 4; /* DP4 */
```

Reply:
thanks @shumingfan, updated the patch and added you as co-author. please check if this is ok now.